doc/developer-notes.md

```diff
@@ -742,14 +743,13 @@ logging messages. They should be used as follows:
   attacker to fill up storage. Note that `LogPrintf(fmt, params...)` is
   a deprecated alias for `LogInfo`.
 
-- `LogError(fmt, params...)` should be used in place of `LogInfo` for
-  severe problems that require the node (or a subsystem) to shut down
-  entirely (e.g., insufficient storage space).
+- `LogError(fmt, params...)` should be used in place of `LogInfo` for severe
+  errors the node admin will need to address (e.g., failure to write data).
 
 - `LogWarning(fmt, params...)` should be used in place of `LogInfo` for
-  severe problems that the node admin should address, but are not
-  severe enough to warrant shutting down the node (e.g., system time
-  appears to be wrong, unknown soft fork appears to have activated).
+  unexpected conditions indicating potentially severe problems the node admin
+  should address (e.g. system time appears to be wrong, unknown soft fork
+  appears to have activated).
 
 - `LogTrace(BCLog::CATEGORY, fmt, params...)` should be used in place of
```

Comment:
nit: the logger order is weird, why debug -> info -> error -> warning -> trace?
Especially after announcing `LogInfo`, `LogDebug`, `LogTrace`, `LogWarning` and `LogError` in the intro.